core/amber/src/main/python/core/runnables/data_processor.py

```diff
@@ -29,40 +30,53 @@ def run(self) -> None:
         self._running.set()
         self._switch_context()
         while self._running.is_set():
-            self.process_tuple()
+            marker = self._context.marker_processing_manager.get_input_marker()
+            tuple_ = self._context.tuple_processing_manager.current_input_tuple
+            if marker is not None:
+                self.process_marker(marker)
+            elif tuple_ is not None:
+                self.process_tuple()
+            else:
+                raise RuntimeError("No marker or tuple to process.")
+            self._switch_context()
+
+    def process_marker(self, marker: Marker) -> None:
+        try:
+            executor = self._context.executor_manager.executor
+            port = self._context.tuple_processing_manager.get_input_port()
+            with replace_print(
+                self._context.worker_id,
+                self._context.console_message_manager.print_buf,
+            ):
```

Comment:
Do we really have the need to support print while handling a state? I suggest we remove it until we really have such need. For now, most of the state handlers (process_state) should be implemented by us.

Reply:
I suggest we support the print. It's really useful while debugging. For example, I can print the state before applying it to the operator to ensure that the state is correct.